lib/base/io-engine.hpp

```diff
@@ -175,6 +175,24 @@ class AsioConditionVariable
 	boost::asio::deadline_timer m_Timer;
 };
 
+/**
+ * Like AsioEvent, but additionally supports waiting for an event to be cleared
+ *
+ * @ingroup base
+ */
+class AsioSymmetricEvent
```

Comment:
As discussed about this yesterday, I personally find this naming doesn't reflect its use cases. Therefore, unless you've a better alternate name, I would rename this to either `AsioDualEvent` or `AsioDualState`. Apart from that, a bit more class docs describing its behaviour would be nice too.